kategory-annotations-processor/src/main/java/kategory/instances/InstanceFileGenerator.kt

```diff
@@ -194,7 +194,7 @@ class InstanceFileGenerator(
     private fun genCompanionReifiedExtensions(i: Instance): String =
             """|
                 |inline fun ${i.expandedTypeArgs(reified = true)} ${i.receiverTypeName}.Companion.${i.companionFactoryName}(${i.args.map {
-                "${it.first}: ${it.second} = ${it.second.split(".").map { it.decapitalize() }.joinToString(".")}()"
+                "${it.first}: ${it.second} = ${it.second.substringBefore("<").split(".").map { it.decapitalize() }.joinToString(".")}<${it.second.substringAfter("<")}()"
```

Comment:
Can you please give a small overview of what this does?

Reply:
I also extracted it to a separate method to make it a little clearer.

Here we are transforming the typeclass type to a method call for resolution.
`kategory.Eq<A>` -> `kategory.eq<A>()` we were doing so by splitting on `.` and decapitalizing.

But this fails for types with fixed types like `Try<A>`.
`kategory.Eq<kotlin.Throwable>` -> `kategory.eq<kotlin.throwable>()`.

On second thought I currently disabled implicit composition of `Eq<Throwable>` for `Try<A>` because this is also doesn't work for resolution. I'll look if this can be added there as well.